htlcswitch/link.go

```diff
@@ -2334,7 +2334,7 @@ func (l *channelLink) processRemoteAdds(fwdPkg *channeldb.FwdPkg,
 			// TODO(conner): track ownership of settlements to
 			// properly recover from failures? or add batch invoice
 			// settlement
-			if invoice.Terms.Settled {
+			if invoice.Terms.State != channeldb.ContractOpen {
```

Comment:
Is this intended? Comment should possibly be updated to reflect change?

Reply:
Yes. In this PR both are equivalent, but I am inverting the condition as a preparation for the hold invoice.

Reply:
Same for comment below